demo-python/code/azure-search-vector-image-index-creation-python-sample.ipynb

```diff
@@ -0,0 +1,389 @@
+    "index_name = os.getenv(\"AZURE_SEARCH_INDEX_NAME\")\n",
+    "key = os.getenv(\"AZURE_SEARCH_ADMIN_KEY\")  \n",
+    "aiVisionApiKey = os.getenv(\"AZURE_AI_VISION_API_KEY\")  \n",
+    "aiVisionRegion = os.getenv(\"AZURE_AI_VISION_REGION\")\n",
+    "blob_connection_string = os.getenv(\"BLOB_CONNECTION_STRING\")  \n",
+    "container_name = os.getenv(\"BLOB_CONTAINER_NAME\")\n",
+    "credential = AzureKeyCredential(key)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Image Embedding\n",
+    "\n",
+    "Vectorize images with image embedding API "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_image_vector(image_path, key, region):\n",
+    "    headers = {\n",
+    "        'Ocp-Apim-Subscription-Key': key,\n",
+    "    }\n",
+    "\n",
+    "    params = urllib.parse.urlencode({\n",
+    "        'model-version': '2023-04-15',\n",
+    "    })\n",
+    "\n",
+    "    MAX_RETRIES = 5\n",
+    "\n",
+    "    for attempt in range(MAX_RETRIES):\n",
+    "        try:\n",
+    "            if image_path.startswith(('http://', 'https://')):\n",
+    "                headers['Content-Type'] = 'application/json'              \n",
+    "                body = json.dumps({\"url\": image_path})\n",
+    "            else:\n",
+    "                headers['Content-Type'] = 'application/octet-stream'\n",
+    "                with open(image_path, \"rb\") as filehandler:\n",
+    "                    image_data = filehandler.read()\n",
+    "                    body = image_data\n",
+    "\n",
+    "            conn = http.client.HTTPSConnection(f'{region}.api.cognitive.microsoft.com', timeout=3)\n",
+    "            conn.request(\"POST\", \"/computervision/retrieval:vectorizeImage?api-version=2023-04-01-preview&%s\" % params, body, headers)\n",
+    "            response = conn.getresponse()\n",
+    "            data = json.load(response)\n",
+    "            conn.close()\n",
+    "\n",
+    "            if response.status != 200:\n",
+    "                print(f\"Error processing image {image_path}: {data.get('message', '')}\")\n",
+    "                return None\n",
+    "\n",
+    "            return data.get(\"vector\")\n",
+    "\n",
+    "        except (requests.exceptions.Timeout, http.client.HTTPException) as e:\n",
+    "            if attempt < MAX_RETRIES - 1:\n",
+    "                print(f\"Timeout/Error for {image_path}. Retrying {attempt+1}/{MAX_RETRIES}...\")\n",
+    "                continue\n",
+    "            else:\n",
+    "                print(f\"Error after {MAX_RETRIES} attempts for {image_path}.\")\n",
+    "                return None\n",
+    "\n",
+    "input_json_file = '../data/images/apples/input.json'\n",
+    "output_folder = '../data/images/apples/output'\n",
+    "if os.path.exists(output_folder):\n",
```

Comment:
Seems like the output will not change if you run it again. Maybe we can skip re-vectorzing the images if the output directory already exists? We have adopted a similar strategy in our existing samples to save on cost

Reply:
If the user wants to re-vectorize , they can manually delete the output directory